crates/rust-client/src/transactions/mod.rs

```diff
@@ -327,31 +328,26 @@ impl<N: NodeRpcClient, R: FeltRng, S: Store, A: TransactionAuthenticator> Client
         maybe_await!(TransactionResult::new(executed_transaction, screener, future_notes))
     }
 
-    /// Proves the specified transaction witness, and returns a [ProvenTransaction] that can be
-    /// submitted to the node.
-    pub fn prove_transaction(
-        &mut self,
-        executed_transaction: ExecutedTransaction,
-    ) -> Result<ProvenTransaction, ClientError> {
-        let transaction_prover = TransactionProver::new(ProvingOptions::default());
-
-        let proven_transaction = transaction_prover.prove_transaction(executed_transaction)?;
-        Ok(proven_transaction)
-    }
-
-    /// Submits a [ProvenTransaction] to the node, and stores the transaction in
+    ///  Proves the specified transaction witness, submits a [ProvenTransaction] to the node, and stores the transaction in
     /// the local database for tracking.
     pub async fn submit_transaction(
         &mut self,
         tx_result: TransactionResult,
-        proven_transaction: ProvenTransaction,
     ) -> Result<(), ClientError> {
+        let start_proving = Instant::now();
+        let transaction_prover = TransactionProver::new(ProvingOptions::default());
+        info!("Proving took: {}ms", start_proving.elapsed().as_millis());
+
+        let proven_transaction = transaction_prover.prove_transaction(tx_result.executed_transaction().clone())?;
+
+        let start_submitting = Instant::now();
         self.rpc_api.submit_proven_transaction(proven_transaction).await?;
-        info!("Transaction submitted");
+        info!("Submitting took: {}ms", start_submitting.elapsed().as_millis());
 
         // Transaction was proven and submitted to the node correctly, persist note details and update account
+        let start_store = Instant::now();
         maybe_await!(self.store.apply_transaction(tx_result))?;
-        info!("Transaction stored");
+        info!("Storing took: {}ms", start_store.elapsed().as_millis());
```

Comment:
nit: 
```suggestion
        info!("Storing transaction data took: {}ms", start_store.elapsed().as_millis());
```

We could also remove this if preferred, I don't mind either way